language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs

```diff
@@ -325,3 +385,84 @@ callProcessSilent cmd args = do
       hPutStrLn stderr $ unlines ["stdout:", out]
       hPutStrLn stderr $ unlines ["stderr: ", err]
       exitFailure
+
+writeTsConfig :: FilePath -> IO ()
+writeTsConfig dir = writeFileUTF8 (dir </> "tsconfig.json") $ unlines
+    [ "{"
+    , "  \"compilerOptions\": {"
+    , "    \"target\": \"es5\","
+    , "    \"lib\": ["
+    , "      \"dom\","
+    , "      \"es2015\""
+    , "     ],"
+    , "    \"strict\": true,"
+    , "    \"noUnusedLocals\": true,"
+    , "    \"noUnusedParameters\": false,"
+    , "    \"noImplicitReturns\": true,"
+    , "    \"noFallthroughCasesInSwitch\": true,"
+    , "    \"outDir\": \"lib\","
+    , "    \"module\": \"commonjs\","
+    , "    \"declaration\": true,"
+    , "    \"sourceMap\": true"
+    , "    },"
+    , "  \"include\": [\"src/**/*.ts\"],"
+    , "}"
+    ]
+
+writePackageJson :: FilePath -> String -> IO ()
+writePackageJson dir name = writeFileUTF8 (dir </> "package.json") $ unlines
+        ["{"
+        , "  \"private\": true,"
+        , "  \"name\": \"" <> name <> "\","
+        , "  \"version\": \"0.13.53\","
+        , "  \"description\": \"Produced by daml2ts\","
+        , "  \"license\": \"Apache-2.0\","
+        , "  \"dependencies\": {"
+        , "    \"@daml/types\": \"0.13.53\","
+        , "    \"@mojotech/json-type-validation\": \"^3.1.0\""
+        , "  },"
+        , "  \"scripts\": {"
+        , "    \"build\": \"tsc --build\","
+        , "    \"build:watch\": \"tsc --build --watch\","
+        , "    \"test\": \"true\","
+        , "    \"lint\": \"eslint --ext .js,.ts src/ --max-warnings 0\""
+        , "  },"
+        , "  \"eslintConfig\": {"
+        , "    \"extends\": ["
+        , "      \"eslint:recommended\","
+        , "      \"plugin:@typescript-eslint/eslint-recommended\","
+        , "      \"plugin:@typescript-eslint/recommended\","
+        , "      \"plugin:@typescript-eslint/recommended-requiring-type-checking\""
+        , "    ],"
+        , "    \"parser\": \"@typescript-eslint/parser\","
+        , "    \"parserOptions\": {"
+        , "      \"project\": \"./tsconfig.json\""
+        , "    },"
+        , "    \"plugins\": ["
+        , "      \"@typescript-eslint\""
+        , "    ],"
+        , "    \"rules\": {"
+        , "      \"@typescript-eslint/no-floating-promises\": \"error\","
+        , "      \"@typescript-eslint/explicit-function-return-type\": \"off\","
+        , "      \"@typescript-eslint/no-inferrable-types\": \"off\""
+        , "    }"
+        , "  },"
+        , "  \"devDependencies\": {"
+        , "    \"@typescript-eslint/eslint-plugin\": \"^2.11.0\","
+        , "    \"@typescript-eslint/parser\": \"^2.11.0\","
+        , "    \"eslint\": \"^6.7.2\","
+        , "    \"typescript\": \"~3.7.3\""
+        , "  }"
+        , "}"
+        ]
+
+writeRootPackageJson :: FilePath -> String -> IO ()
+writeRootPackageJson rootDir workspace =
```

Comment:
What do we need this one for?

Reply:
We don't. Removing.

Reply:
Actually, turns out we do. Need to write one for the `yarn install` step to work but otherwise don't if we `cd x&& yarn build`.

Reply:
Introducing linting I've uncovered another reason to do so. To make it so that we can `yarn install` only once, we really want to `yarn workspaces run lint` if we don't want to have to `yarn install` every package for installing `eslint` (which costs 10 seconds or so for each test).

Reply:
Is there a way to depend on the "global" `node_modules` directory we maintain with bazel? @aherrmann do you know the answer?
If there is, we might get away without using workspaces. If we have to use workspaces, I'd prefer to call `yarn build` and `yarn lint` from the directory containing the generated code rather than calling the `yarn workspaces run ...` versions, which also run over the `@dummy/dummy` package.

Reply:
@hurryabit You can depend on `"@language_support_ts_deps//:node_modules"` to put them into the sandbox. I'm not sure if there is a way to point yarn to another `node_modules` folder, if not you could create a symlink to put it in the right place.

Alternatively, you could turn the `yarn lint` call into it's own `eslint_test` target. However, that would require some larger refactoring of this PR.